spec/models/card_spec.rb

```diff
@@ -1,188 +1,328 @@
-                       user_id: 1)
-    expect(card.errors[:block_id]).
-        to include('Выберите колоду из выпадающего списка.')
-  end
+      it 'returns attribute equal to true' do
+        expect(check_result[:state]).to be true
+      end
 
-  it 'check_translation Eng OK levenshtein_distance' do
-    card = Card.create(original_text: 'дом', translated_text: 'hous',
-                       user_id: 1, block_id: 1)
-    check_result = card.check_translation('house')
-    expect(check_result[:state]).to be true
-  end
+      it 'returns attribute equal to 1' do
+        expect(check_result[:distance]).to eq(1)
+      end
+    end
 
-  it 'check_translation Eng OK levenshtein_distance=1' do
-    card = Card.create(original_text: 'дом', translated_text: 'hous',
-                       user_id: 1, block_id: 1)
-    check_result = card.check_translation('house')
-    expect(check_result[:distance]).to be 1
-  end
+    context 'when translation is incorrect' do
+      let(:check_result) { card.check_translation('RoR') }
 
-  it 'check_translation Rus OK levenshtein_distance' do
-    card = Card.create(original_text: 'house', translated_text: 'до',
-                       user_id: 1, block_id: 1)
-    check_result = card.check_translation('дом')
-    expect(check_result[:state]).to be true
-  end
+      it 'returns state attribute equal to false' do
+        check_result = card.check_translation('RoR')
+        expect(check_result[:state]).to be false
+      end
 
-  it 'check_translation Rus OK levenshtein_distance=1' do
-    card = Card.create(original_text: 'house', translated_text: 'до',
-                       user_id: 1, block_id: 1)
-    check_result = card.check_translation('дом')
-    expect(check_result[:distance]).to be 1
+      it 'returns distance attribute > 1' do
+        check_result = card.check_translation('RoR')
+        expect(check_result[:distance]).to be > 1
+      end
+    end
   end
 
-  it 'check_translation Eng NOT levenshtein_distance=2' do
-    card = Card.create(original_text: 'дом', translated_text: 'hou',
-                       user_id: 1, block_id: 1)
-    check_result = card.check_translation('RoR')
-    expect(check_result[:state]).to be false
-  end
+  describe "#check translation sets correct card attributes" do
+    before do
+      card.save
+    end
+
+    def check_translation(translation, attempt=1)
+      attempt.times do
+        card.check_translation(translation)
+        card.reload
+      end
+    end
+
+    def expect_parameters(parameters={})
+      parameters.each_pair do |key, value|
+        expect(card[key]).to eq(value)
+      end
+    end
+
+    context 'with correct translation' do
+      
+      describe 'when interval: 1, repeat: 1, efactor: 2.5, quality: 5' do
+        before(:each) do
+          card.update_attributes(interval: 1, repeat: 1, efactor: 2.5, quality: 5)
+        end
+
+        it 'set correct review date, interval, repeat and attempt for repeat=1 quality=5' do
```

Comment:
Line is too long. [92/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.